test/agent.ts

```diff
@@ -29,7 +30,8 @@ const prepareServer = server => {
 	});
 };
 
-const createAgentSpy = Cls => {
+// TODO: Type this stricter if possible
+const createAgentSpy = <T extends any>(Cls: T) => {
```

Comment:
```suggestion
const createAgentSpy = <T>(Cls: T) => {
```

And let's use a clearer name than `Cls`.

Reply:
Hmm. `unknown` doesn't seems to work here because it has to be something with a construct-able signature. Since `unknown` is not callable, it can never fulfill the type requirements?

Reply:
Ah, but it should at least be something more specific than `any` then.